aten/src/ATen/cuda/CUDASolver.h

```diff
@@ -77,6 +82,214 @@ void gesvdjBatched<c10::complex<float>>(CUDASOLVER_GESVDJ_BATCHED_ARGTYPES(c10::
 template<>
 void gesvdjBatched<c10::complex<double>>(CUDASOLVER_GESVDJ_BATCHED_ARGTYPES(c10::complex<double>, double));
 
+#define CUDASOLVER_SYEVD_BUFFERSIZE_ARGTYPES(scalar_t, value_t)             \
+  cusolverDnHandle_t handle, cusolverEigMode_t jobz, cublasFillMode_t uplo, \
+      int n, const scalar_t *A, int lda, const value_t *W, int *lwork
+
+template <class scalar_t, class value_t = scalar_t>
+void syevd_bufferSize(CUDASOLVER_SYEVD_BUFFERSIZE_ARGTYPES(scalar_t, value_t)) {
+  TORCH_CHECK(
```

Comment:
Should this be TORCH_CHECK? If so, is there a user-facing name instead of the C++ name it should use?

Reply:
This is not a user-facing error, only for developers. This could be replaced with `TORCH_INTERNAL_ASSERT(false);`.